moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -541,20 +533,22 @@ double JogCalcs::decelerateForSingularity(const Eigen::VectorXd& commanded_veloc
   return velocity_scale;
 }
 
-void JogCalcs::enforceJointVelocityLimits(Eigen::VectorXd& calculated_joint_change)
+bool JogCalcs::checkIfJointsWithinSRDFBounds(Eigen::ArrayXd& calculated_joint_change)
 {
-  double maximum_joint_change = calculated_joint_change.cwiseAbs().maxCoeff();
-  if (maximum_joint_change > parameters_.joint_scale * parameters_.publish_period)
+  for (auto joint : joint_model_group_->getJointModels())
   {
-    // Scale the entire joint velocity vector so that each joint velocity is below min, and the output movement is
-    // scaled uniformly to match expected motion
-    calculated_joint_change =
-        calculated_joint_change * parameters_.joint_scale * parameters_.publish_period / maximum_joint_change;
+    // Some joints do not have bounds defined
+    if (kinematic_state_->getJointModel(joint->getName())->hasVariable(joint->getName()))
+    {
+      auto bounds = kinematic_state_->getJointModel(joint->getName())->getVariableBounds(joint->getName());
+      ROS_ERROR_STREAM(bounds.min_velocity_);
+
+      // Apply velocity bounds
+    }
   }
-}
+  ROS_WARN_STREAM("===");
 
-bool JogCalcs::checkIfJointsWithinURDFBounds(trajectory_msgs::JointTrajectory& new_joint_traj)
-{
+/*
```

Comment:
Please add a NOTE/TODO why this is commented

Reply:
Yeah, this is Work In Progress. Which I think we discussed reverting, submitting the basics of this PR, then adding a followup PR. Sorry for not marking WIP